aws/resource_aws_workspace.go

```diff
@@ -0,0 +1,405 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsWorkspace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWorkspaceCreate,
+		Read:   resourceAwsWorkspaceRead,
+		Update: resourceAwsWorkspaceUpdate,
+		Delete: resourceAwsWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"bundle_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"directory_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"computer_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"root_volume_encryption_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+			"user_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_volume_encryption_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+			"volume_encryption_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"workspace_properties": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
```

Comment:
Done. Just curious, is `Computed` option only for documentation purpose? How could it imply at resource lifecycle?

Reply:
@gdavison Please could you elaborate on `Computed` property?

Reply:
`Computed` values are populated in the resource by the `Read` operation. If a value has only the `Computed` property, it is output-only. If it is `Optional` and `Computed`, then you can set a value, and the resource will use it, otherwise it will return the values that AWS populates the field with.

Reply:
Thank you very much for the explanation, it's crystal clear 💎